src/bridge/setup.rs

```diff
@@ -100,6 +108,11 @@ pub async fn setup_neovide_specific_state(
         .await
         .context("Error when running Neovide init.lua")?;
 
+    #[cfg(target_os = "macos")]
+    setup_startup_directory(nvim)
```

Comment:
Could we move this to the command implementation instead? Something like `cd $HOME;nvim` but a little bit more advanced to deal with all the arguments.

This way any arguments that changes the startup directory for `nvim` works correctly, for example `neovide -- -c "cd proj"`

Reply:
that was a good idea! done.

Reply:
@fredizzimo the idea is good but actually that implementation was wrong.

any neovim command passing through neovide should be executed already when we launch the `--embed` instance so 1. it would be duplicated and 2. it should be made by the command line parser as it is right now.

I also forgot that some TTY could pass nvim commands such as the startup directory (hello Neohub), so as it was implemented, those TTY would break.

I think it's better to ignore *any* neovim command and just handle the startup path received in case there is one. If not we understand the the dafault is $HOME.

Reply:
@fredizzimo done some refactor to make it work respecting the `--cmd` or `-c` nvim commands.

Executing it in lua was the only way to make it respect the startup directories and the priority of commands tho.

I've tested using Finder, Dock and Neohub. (the terminal sessions shouldn't be affected)